pkg/storage/stores/shipper/compactor/retention/expiration.go

```diff
@@ -113,27 +120,33 @@ Outer:
 	return globalRetention
 }
 
-func findSmallestRetentionPeriod(limits Limits) time.Duration {
+func findLatestRetentionStartTime(now model.Time, limits Limits) (model.Time, map[string]model.Time) {
+	// find the smallest retention period from default limits
```

Comment:
Seems like it could be a function comment instead.

Reply:
I wanted to explain the block below that comment. I will add a function comment as well.